File: iteration-two-task2/frontend/src/api/task.ts
import client from "./client";

export interface TaskConfig {
    metrics: string[];
    strategy: string;
    enable_process_trace: boolean;
}

export interface TaskEntity {
    id: number;
    name: string;
    description?: string;
    agent_id?: number | null;
    agent_version: string;
    dataset_id: string;
    mode: string;
    eval_mode?: string;
    method: string;
    dimension: string;
    status: string;
    config: TaskConfig;
    judge_config?: Record<string, unknown>;
    run_config?: Record<string, unknown>;
    metrics?: string[];
    input_payload: Record<string, unknown>;
    progress?: number;
    total_samples?: number;
    completed_samples?: number;
    failed_samples?: number;
    started_at?: string | null;
    finished_at?: string | null;
    error_message?: string | null;
    note?: string;
    created_at: string;
    updated_at: string;
}

export interface TaskListResponse {
    total: number;
    page: number;
    page_size: number;
    items: TaskEntity[];
}

export const listTasks = async () => {
    const { data } = await client.get<TaskListResponse>("/tasks/", { params: { page_size: 200 } });
    return data;
};

export const createTask = async (payload: Partial<TaskEntity>) => {
    const { data } = await client.post<TaskEntity>("/tasks/", payload);
    return data;
};

export const updateTask = async (taskId: number, payload: Partial<TaskEntity>) => {
    const { data } = await client.put<TaskEntity>(`/tasks/${taskId}`, payload);
    return data;
};

export const deleteTask = async (taskId: number) => {
    await client.delete(`/tasks/${taskId}`);
};

export const executeTask = async (taskId: number) => {
    const { data } = await client.post(`/tasks/${taskId}/execute`);
    return data;
};

export const runTask = async (taskId: number) => {
    const { data } = await client.post(`/tasks/${taskId}/run`);
    return data;
};

export const cancelTask = async (taskId: number) => {
    const { data } = await client.post(`/tasks/${taskId}/cancel`);
    return data;
};

export const cloneTask = async (taskId: number) => {
    const { data } = await client.post<TaskEntity>(`/tasks/${taskId}/clone`);
    return data;
};

export const compareTasks = async (taskIds: number[]) => {
    const { data } = await client.post("/tasks/compare", { task_ids: taskIds });
    return data;
};

export const exportTaskResults = async (taskId: number) => {
    const { data } = await client.get<string>(`/tasks/${taskId}/results/export?format=csv`);
    return data;
};

File: iteration-two-task2/backend/app/schemas/task.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


TaskStatus = Literal["draft", "pending", "running", "completed", "failed", "cancelled"]
EvalMode = Literal["result", "process", "result_and_process", "dataset-batch"]
EvalMethod = Literal["explicit", "fuzzy"]
EvalDimension = Literal["effectiveness", "safety", "performance"]


class TaskConfig(BaseModel):
    metrics: list[str] = Field(default_factory=list)
    strategy: str = "default"
    enable_process_trace: bool = False


class TaskCreate(BaseModel):
    name: str = Field(min_length=2, max_length=120)
    description: str | None = None
    agent_id: int | None = None
    agent_version: str = "v1"
    dataset_id: str
    mode: EvalMode = "result"
    eval_mode: EvalMode | None = None
    method: EvalMethod = "explicit"
    dimension: EvalDimension = "effectiveness"
    config: TaskConfig = Field(default_factory=TaskConfig)
    judge_config: dict = Field(default_factory=dict)
    run_config: dict = Field(default_factory=dict)
    metrics: list[str] = Field(default_factory=list)
    input_payload: dict = Field(default_factory=dict)
    note: str | None = None


class TaskUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    agent_id: int | None = None
    agent_version: str | None = None
    dataset_id: str | None = None
    mode: EvalMode | None = None
    eval_mode: EvalMode | None = None
    method: EvalMethod | None = None
    dimension: EvalDimension | None = None
    status: TaskStatus | None = None
    config: TaskConfig | None = None
    judge_config: dict | None = None
    run_config: dict | None = None
    metrics: list[str] | None = None
    input_payload: dict | None = None
    note: str | None = None


class TaskRead(BaseModel):
    id: int
    name: str
    description: str | None = None
    agent_id: int | None = None
    agent_version: str
    dataset_id: str
    mode: EvalMode
    eval_mode: EvalMode
    method: EvalMethod
    dimension: EvalDimension
    status: TaskStatus
    config: dict
    judge_config: dict
    run_config: dict
    metrics: list[str]
    input_payload: dict
    progress: int
    total_samples: int
    completed_samples: int
    failed_samples: int
    started_at: datetime | None = None
    finished_at: datetime | None = None
    error_message: str | None = None
    note: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TaskListResponse(BaseModel):
    total: int
    page: int
    page_size: int
    items: list[TaskRead]
